addons/enemybar/enemybar.lua

```diff
@@ -24,31 +24,524 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+            worser_match = mob
+        end
+    end
+    return worse_match and worse_match or worser_match
+end
+
+function normalize_bar_name(n)
+    n = n:lower()
+    if n == 't' then
+        n = 'target'
+    elseif n == 'st' then
+        n = 'subtarget'
+    elseif n == 'a' then
+        n = 'aggro'
+    elseif S{'f','ft'}:contains(n) then
+        n = 'focustarget'
+    elseif not S{'target','subtarget','aggro','focustarget','all'}:contains(n) then
+        n = nil
+    end
+
+    return n
+end
+
+function normalize_boolean(b)
+    b = b:lower()
+    if S{'true','t','yes','y','on'}:contains(b) then
+        return true
+    elseif S{'false','f','no','n','off'}:contains(b) then
+        return false
+    else return nil end
+end
+
+function set_setting(bar, setting, v)
+    if bar == 'all' then
+        settings['target_bar'][setting] = v
+        settings['subtarget_bar'][setting] = v
+        settings['aggro_bar'][setting] = v
+        settings['focustarget_bar'][setting] = v
+    else
+        settings[bar..'_bar'][setting] = v
+    end
+    windower.add_to_chat(207, 'EnemyBar: "'..setting..'" updated for "'..bar..'" bar')
+    settings:save()
+    initialize_bars()
+end
+
+function load_settings()
+    defaults = {}
+    defaults.target_bar = {
+        pos={x=650,y=750}, width=600,
+        color={alpha=255,red=255,green=0,blue=0},
+        font='Arial', font_size=14,
+        show=true, show_target=false, show_target_icon=false,
+        show_action=false, show_dist=false, show_debuff=false}
+    defaults.subtarget_bar = {
+        pos={x=680,y=700}, width=300,
+        color={alpha=255,red=12,green=50,blue=101},
+        font='Arial', font_size=12,
+        show=true, show_target=false, show_target_icon=false,
+        show_action=false, show_dist=false, show_debuff=false}
+    defaults.focustarget_bar = {
+        pos={x=680,y=670}, width=250,
+        color={alpha=255,red=93,green=0,blue=255},
+        font='Arial', font_size=12,
+        show=true, show_target=false, show_target_icon=false,
+        show_action=false, show_dist=false, show_debuff=false}
+    defaults.aggro_bar = {
+        pos={x=350,y=550}, width=180,
+        color={alpha=255,red=0,green=150,blue=50},
+        font='Arial', font_size=9,
+        show=false, show_target=false, show_target_icon=false,
+        show_action=false, show_dist=false, show_debuff=false,
+        count=6, stack_dir='down', stack_padding = 27}
+    settings_old = config.load({})
+    if settings_old.pos then
+        -- settings.pos was the old position setting. if it's set that means we're upgrading from 1.0
+        defaults.target_bar.pos = settings_old.pos
+        defaults.target_bar.font = settings_old.font
+        defaults.subtarget_bar.font = settings_old.font
+        defaults.target_bar.font_size = settings_old.font_size
+        defaults.subtarget_bar.font_size = settings_old.font_size
+    end
+    settings = config.load(defaults)
+
+    initialize_bars()
```

Comment:
The `config` library has a feature you can use for this:
```lua
config.register(settings, initialize_bars)
```

This will execute `initialize_bars` every time the settings are reloaded (i.e. once initially at the start, but also when you relog into a different character, with different settings. You can also manually trigger it with `config.reload(settings)`, as you could do above in the `set_setting` function.